crates/pallet-domains/src/lib.rs

```diff
@@ -1482,6 +1493,44 @@ impl<T: Config> Pallet<T> {
             FraudProofError::ChallengingGenesisReceipt
         );
 
+        if let FraudProof::InvalidBundle(InvalidBundleProof::TotalRewards(
```

Comment:
If we want to make the fraud proof stateless and verifiable from the light client, then it should contain the `bad_receipt` by itself along with a storage proof of the `bad_receipt` and the verifier only needs to provide the consensus chain state root.